apps/explorer/src/lib/ui/utils/generateValidatorsTableColumns.tsx

```diff
@@ -257,20 +262,79 @@ export function generateValidatorsTableColumns({
                         </TableCellBase>
                     );
                 }
+
                 return (
                     <TableCellBase>
                         <Badge
                             type={
-                                atRisk === null && isCommitteeMember
-                                    ? BadgeType.PrimarySoft
-                                    : BadgeType.Neutral
+                                atRisk === null
+                                    ? BadgeType.Success
+                                    : atRisk > 1
+                                      ? BadgeType.Warning
+                                      : BadgeType.Error
                             }
                             label={label}
                         />
                     </TableCellBase>
                 );
             },
         },
+        {
+            header: 'Current Epoch Rewards',
+            accessorKey: 'isEarningCurrent',
+            id: 'isEarningCurrent',
+            enableSorting: true,
+            cell({ row }) {
+                const isCommitteeMember = committeeMembers.find(
+                    (committeeMemberAddress) => committeeMemberAddress === row.original.iotaAddress,
+                );
+                const label = isCommitteeMember ? 'Earning' : 'Not Earning';
+                return (
+                    <TableCellBase>
+                        <Badge
+                            type={isCommitteeMember ? BadgeType.PrimarySoft : BadgeType.Neutral}
+                            label={label}
+                        />
+                    </TableCellBase>
+                );
+            },
+        },
+        {
+            header: 'Next Epoch Rewards',
+            accessorKey: 'isEarningNext',
+            id: 'isEarningNext',
+            enableSorting: true,
+            cell({ row }) {
+                const { atRisk, isPending, isCommitteeMember } = determineRisk(
+                    committeeMembers,
+                    atRiskValidators,
+                    row,
+                );
+
+                const isInTopStakers = !!topValidators.find(
+                    (v) => v.iotaAddress === row.original.iotaAddress,
+                );
+
+                // if its active or pending validator, not at high risk (high risk, not normal risk),
+                // and is part of the top X stakers,
+                // it will generate rewards, otherwise not.
+                const isEarningNext =
+                    (isPending || isCommitteeMember) &&
```

Comment:
shouldnt this be based on if its pending or active? doesnt need to be a committee member, just needs to be active or pending

Reply:
sorry my bad, all validators in this context are active or pending, so it should be something like
```ts
// if its active or pending validator (all validators in this context are either active or pending),
// not at high risk (high risk, not normal risk),
// and is part of the top X stakers,
// it will generate rewards, otherwise not.
const isEarningNext =
    (atRisk === null || atRisk > 1) &&
    isInTopStakers;
```

Reply:
[aa8a61c](https://github.com/iotaledger/iota/pull/6601/commits/aa8a61cc513c04d12b370b17ebaeecf82336ef2c)